### PR TITLE
Update for SMP support

### PR DESCRIPTION
**Pull request by UncleGrumpy**
Updates the driver for SMP support in AtomVM `master` builds using ESP-IDF v4.x. Due to race conditions created by smp `wifi_acquire` and `wifi_relese` have been removed, since there is no way to fake support for these functions either, until the native esp32 wifi driver can be fully stopped by the AtomVM network driver.

This PR does not include support for building with ESP-IDF v5.x.

### CODE REVIEW

File: nifs/atomvm_adc.c [hunk 1]
```diff
@@ -15,93 +15,71 @@
...
+    { ATOM_STR("\x6", "db_11"), ADC_ATTEN_DB_11 },
+    SELECT_INT_DEFAULT(ADC_ATTEN_MAX)
+};
 
-static const char *const samples_atom         = "\x7"  "samples";
-static const char *const raw_atom             = "\x3"  "raw";
-static const char *const voltage_atom         = "\x7"  "voltage";
+static const char *const samples_atom = "\x7" "samples";
+static const char *const raw_atom     = "\x3" "raw";
+static const char *const voltage_atom = "\x7" "voltage";
```
Comment (fadushin) [inline nifs/atomvm_adc.c]: Can we use the ATOM_STR macro here, while we are at it?

File: nifs/atomvm_adc.c [hunk 2]
```diff
@@ -15,93 +15,71 @@
...
 
-static const char *const invalid_pin_atom     = "\xb"  "invalid_pin";
-static const char *const invalid_width_atom   = "\xd"  "invalid_width";
-static const char *const invalid_db_atom      = "\xa"  "invalid_db";
+static const char *const invalid_pin_atom   = "\xb" "invalid_pin";
+static const char *const invalid_width_atom = "\xd" "invalid_width";
+static const char *const invalid_db_atom    = "\xa" "invalid_db";
```
Comment (fadushin) [inline nifs/atomvm_adc.c]: ditto

File: nifs/atomvm_adc.c [hunk 3]
```diff
@@ -432,51 +395,50 @@ static term nif_adc_take_reading(Context *ctx, int argc, term argv[])
...
             return error_tuple;
         }
     }
 
     term read_options = argv[1];
     VALIDATE_VALUE(read_options, term_is_list);
     term samples = interop_proplist_get_value_default(
-        read_options, context_make_atom(ctx, samples_atom), term_from_int(DEFAULT_SAMPLES)
-    );
+        read_options, globalcontext_make_atom(ctx->global, samples_atom), term_from_int(DEFAULT_SAMPLES));
```
Comment (fadushin) [inline nifs/atomvm_adc.c]: We can use `interop_kv_get_value_default` without having to make an atom.  Let's use that here.

File: nifs/atomvm_adc.c [hunk 4]
```diff
@@ -432,51 +395,50 @@ static term nif_adc_take_reading(Context *ctx, int argc, term argv[])
...
     avm_int_t samples_val = term_to_int(samples);
     TRACE("take_reading samples: %i\n", samples_val);
     term raw = interop_proplist_get_value_default(
-        read_options, context_make_atom(ctx, raw_atom), FALSE_ATOM
-    );
+        read_options, globalcontext_make_atom(ctx->global, raw_atom), FALSE_ATOM);
```
Comment (fadushin) [inline nifs/atomvm_adc.c]: ditto

File: nifs/atomvm_adc.c [hunk 5]
```diff
@@ -432,51 +395,50 @@ static term nif_adc_take_reading(Context *ctx, int argc, term argv[])
...
     term voltage = interop_proplist_get_value_default(
-        read_options, context_make_atom(ctx, voltage_atom), FALSE_ATOM
-    );
+        read_options, globalcontext_make_atom(ctx->global, voltage_atom), FALSE_ATOM);
```
Comment (fadushin) [inline nifs/atomvm_adc.c]: ditto

**Review by fadushin — COMMENTED** [review]
Just a few minor suggestions.  Thanks!!

File: nifs/atomvm_adc.c (hunk 1 above)
Comment (UncleGrumpy) [reply nifs/atomvm_adc.c]: Yes! My mistake... I had added them, and then ran clang-format in place over the file, which did some nasty things to those lines. So I used git to revert then, and it looks like I forgot to commit those changes before running the format...

Comment (fadushin) [reply nifs/atomvm_adc.c]: Actually, I was just asking if you could use, for example,
```
ATOM_STR("\x7", "voltage")
```

which would address the reformatting stuff that happens to me, too.

**Comment by UncleGrumpy** [comment]
I lifted your `static term create_pair()` from the spi_driver and used it along with your suggestions. Also did a little reordering to match the style guide better, but I may not have the headers arranged correctly, I alphabetized the separate source locations (i.e. AtomVM libs separate from IDF libs, but each source group is alphabetized).

**Review by fadushin — APPROVED** [review]

**Pull request merged**